app/helpers/blog_post_helper.rb

```diff
@@ -6,7 +6,7 @@ module BlogPostHelper
   include BlogLanguages
 
   def blog_post_tags(post)
-    raw post.tag_list.map { |tag| link_to(tag, filter_posts_path(tag)) }.join(", ")
+    raw post.tag_list.map { |tag| link_to(tag, filter_posts_path(clean_up_tag(tag))) }.join(", ")
```

Comment:
you should use a custom parser instead, see https://github.com/mbleigh/acts-as-taggable-on/blob/master/lib/acts_as_taggable_on/generic_parser.rb